energy-integration/fees-collector/src/fees_accumulation.rs

```diff
@@ -21,32 +20,47 @@ pub trait FeesAccumulationModule:
             "Only known contracts can deposit"
         );
 
-        let (payment_token, payment_amount) = self.call_value().single_fungible_esdt();
+        let payment = self.call_value().single_esdt();
         require!(
-            self.known_tokens().contains(&payment_token),
+            self.known_tokens().contains(&payment.token_identifier),
             "Invalid payment token"
         );
 
         let current_week = self.get_current_week();
-        self.accumulated_fees(current_week, &payment_token)
-            .update(|amt| *amt += &payment_amount);
+        if payment.token_nonce == 0 {
+            self.accumulated_fees(current_week, &payment.token_identifier)
+                .update(|amt| *amt += &payment.amount);
+        } else {
+            self.accumulated_locked_fees(current_week, &payment.token_identifier)
```

Comment:
you need token_identifier and NONCE

Reply:
The storage contains `EsdtTokenPayment`. I only keep the (week, token_id) to iterate through `accumulated_locked_fees`. When needed, I extract the nonce and amount from EsdtTokenPayment.